---
title: "DESEQ2_Chrys_soil_v_hydro"
author: "Silas and Yusuf"
date: "2025-12-02"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Libraries 
```{r}
library(DESeq2)
library(dplyr)
library(tidyr)
library(ggplot2)
library(scales)
library(ggpubr)
library(vsn)  
library("pheatmap")
library("vsn")
```



## Make metadata - coldata df for DESeq2
```{r}
coldata<-data.frame(
  row.names = c('SRR15057665','SRR15057666','SRR15057667','SRR15057668','SRR15057669','SRR15057670'),
  condition = c('hydro','hydro','hydro','soil','soil','soil'),
  replicate = c('1','2','3','1','2','3')
)
```

```{r}
# Check out the data see what we need for deseq2
getwd()

colnames(counts)
rownames(counts)

```


```{r}
library(DESeq2)

setwd('/gpfs1/home/s/m/smdecker/projects/eco_genomics_2025/transcriptomics/mydata')
# Load counts
counts <- read.table("STAR_counts_matrix_clean.tsv",
                     header = TRUE,
                     sep = "\t",
                     row.names = 1)

# Make sure names match
stopifnot(all(colnames(counts) == rownames(coldata)))

# Build DESeq2 object
dds <- DESeqDataSetFromMatrix(countData = counts,
                              colData = coldata,
                              design = ~ condition)

dds <- DESeq(dds)

res <- results(dds)

resultsNames(dds)

dim(dds)

# How many transcripts have 15+ counts in 2/3 samples
# a lot: 21,534
ddsQC1 <- dds[rowSums(counts(dds) >= 15) >= 4,]
nrow(ddsQC1)

write.csv(as.data.frame(res), "DESeq2_results.csv")

#check that our data is sane 
#all(colnames(counts) == rownames(coldata))
```

## Quick characterization of data - need to clean these up 
```{r}

# Let's see how many reads we have from each sample
colSums(counts)
mean(colSums(counts))

barplot(colSums(counts), names.arg=colnames(counts),cex.names=0.5, las=3,ylim=c(0,21000000)) #change dims
barplot
abline(h=mean(colSums(counts)), col="blue", lwd=2)

# the average number of counts per gene
rowSums(counts)
mean(rowSums(countsTableRound)) # 
median(rowSums(countsTableRound)) #

apply(countsTableRound,2,mean) # 2 in the apply function does the action across columns
apply(countsTableRound,1,mean) # 1 in the apply function does the action across rows
hist(apply(counts,1,mean),xlim=c(0,10000), ylim=c(0,60000),breaks=10000)
```



## Transform to remove dependence of variance on the mean 
```{r}
# this gives log2(n + 1)
ntd <- normTransform(dds)
meanSdPlot(assay(ntd))

# Variance stabilizing transformation - vsd is variance stablized dist
vsd <- vst(dds, blind=FALSE)
meanSdPlot(assay(vsd))

```



## Heatmap of dissimilarity of samples
```{r}
sampleDists <- dist(t(assay(vsd)))

library("RColorBrewer")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$condition, vsd$replicate, sep="-")
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```
```{r}
library(pheatmap)
sampleDists <- dist(t(assay(vsd)))
pheatmap(as.matrix(sampleDists), clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists)

```
```{r}
library(pheatmap)

top_genes <- head(order(res$padj, na.last=NA), 50)
mat <- assay(vsd)[top_genes, ]
mat <- mat - rowMeans(mat)

pheatmap(mat, show_rownames=FALSE,
         annotation_col=coldata)

```



## PCA 

```{r}
# first transform the data for plotting using variance stabilization
vsd <- vst(dds, blind=FALSE)

pcaData <- plotPCA(vsd, intgroup=c("condition","replicate"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData,"percentVar"))

ggplot(pcaData, aes(PC1, PC2, color=condition, shape=replicate)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()
```
### Prettier PCA
```{r}

# Variance stabilizing transform for PCA
vsd <- vst(dds, blind = FALSE)

# Extract PCA data
pcaData <- plotPCA(vsd, intgroup = c("condition", "replicate"), returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

# PCA plot for your experiment
ggplot(pcaData, aes(PC1, PC2)) +
  geom_point(aes(color = condition, shape = replicate), size = 4, stroke = 1) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  theme_bw() +
  theme(
    panel.border = element_rect(color = "black", fill = NA, size = 2),
    text = element_text(size = 18)
  ) +
  scale_color_manual(values = c("hydro" = "#6699CC", "soil" = "#CC3333"))


```

### SRR15057665	hydroponic 1 
### SRR15057666	hydroponic 2 
### SRR15057667	hydroponic 3
### SRR15057668	soil 1 
### SRR15057669	soil 2 
### SRR15057670	soil 3 


## Making Groups - Maybe I think we mainly care about condition (SvH) 

```{r}


```

## Visualization 

### How do I find top DE genes?

```{r}
# res is the results from dds

# order by largest adjusted p-value
res_ordered <- res[order(res$padj), ]

head(res_ordered, 20)

```


## Plot Individual Genes 


```{r}

# hmm this does not seem to plot, but holy cow does it show DE of this gene 
d <-plotCounts(dds, gene="Cse_sc000001.1_g220.1", intgroup = (c("condition","replicate")), returnData=TRUE)
d


```

### Most up-regulated genes

```{r}
# first clean out NAs

res_clean <- res[!is.na(res$padj), ]
res_sorted <- res_clean[order(res_clean$padj), ]

head(res_sorted, 10)
```

## MA plot 
```{r}
plotMA(res_sorted, ylim=c(-9,7))
```



## Volcano Plot 
```{r}
# Make a dataframe
res_df <- as.data.frame(res_sorted)

# Remove NAs I already did this 
#res_df <- res_df[!is.na(res_df$padj), ]

# Add significance column for color mapping
res_df$significance <- "Not Sig"
res_df$significance[res_df$padj < 0.05 & res_df$log2FoldChange > 0] <- "Up"
res_df$significance[res_df$padj < 0.05 & res_df$log2FoldChange < 0] <- "Down"

ggplot(res_df, aes(x = log2FoldChange, y = -log10(padj), color = significance)) +
  geom_point(alpha = 0.7, size = 2) +
  scale_color_manual(values = c("Up" = "red", "Down" = "blue", "Not Sig" = "grey")) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "black") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "black") +
  labs(
    title = "Volcano Plot: LFC v Significance",
    x = expression(log[2]~Fold~Change),
    y = expression(-log[10]~Adjusted~p~value)
  ) +
  theme_bw(base_size = 16) +
  theme(legend.title = element_blank())


```


```{r}
topgenes <- head(rownames(res_sorted),20)
mat <- assay(vsd)[topgenes,]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(dds)[,c("condition","replicate")])
pheatmap(mat, annotation_col=df)
pheatmap(mat, annotation_col=df, cluster_cols = F)
```

```{r}
top20<-head(res_sorted,21)

top10_gene_ids <- rownames(head(res_sorted, 10))

head(top20, 21)
```

```{r}
head(topgenes,20)
```

## Explore Top DE Genes


```{r}
topgenes30 <- head(rownames(res_sorted),30)
mat <- assay(vsd)[topgenes30,]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(dds)[,c("condition","replicate")])
pheatmap(mat, annotation_col=df)
pheatmap(mat, annotation_col=df, cluster_cols = F)
```


